### PR TITLE
Powerdns recursor

### PR DESCRIPTION
**Pull request by l2isbad**
fixes #3970

yo @lwhitworth

One more thing to do (102nd), please test it :smile_cat:

### CODE REVIEW

**Comment by lwhitworth** [comment]
Testing now (figured it's the last I could do).  Seeing this though:

```
bash-4.2$ /usr/libexec/netdata/plugins.d/python.d.plugin trace 1 powerdns
2018-07-24 14:10:36: python.d INFO: plugin: main: Using python 2
2018-07-24 14:10:36: python.d ERROR: powerdns: job_name: Url: http://127.0.0.1:8082/api/v1/servers/localhost/statistics. Error: 'NoneType' object has no attribute 'request'
2018-07-24 14:10:36: python.d ERROR: powerdns: job_name: Traceback (most recent call last):
  File "/usr/libexec/netdata/python.d/python_modules/bases/FrameworkServices/UrlService.py", line 95, in _get_raw_data
    status, data = self._get_raw_data_with_status(url, manager)
  File "/usr/libexec/netdata/python.d/python_modules/bases/FrameworkServices/UrlService.py", line 113, in _get_raw_data_with_status
    response = manager.request(method='GET',
AttributeError: 'NoneType' object has no attribute 'request'
2018-07-24 14:10:36: python.d INFO: powerdns: job_name: check() => [FAILED]
2018-07-24 14:10:36: python.d INFO: plugin: main: FINISHED
```

**Comment by l2isbad** [comment]
Ops, will fix it asap

**Comment by l2isbad** [comment]
@lwhitworth sorry. It's fixed now.

File: python.d/powerdns.chart.py [hunk 1]
```diff
@@ -45,15 +46,95 @@
...
 class Service(UrlService):
     def __init__(self, configuration=None, name=None):
         UrlService.__init__(self, configuration=configuration, name=name)
         self.order = ORDER
         self.definitions = CHARTS
 
+    def check(self):
+        self._manager = self._build_manager()
+        if not self._manager:
+            return None
```
Comment (paulfantom) [inline python.d/powerdns.chart.py]: Why not:

Comment (paulfantom) [inline python.d/powerdns.chart.py]: Why not:
```python
if not self._build_manager():
    return None
```
?

Comment (l2isbad) [reply python.d/powerdns.chart.py]: We have to assign a value to `self._manager ` (it makes sense if `self._build_manager()` doesn't fail)

**Comment by lwhitworth** [comment]
Spot on.  Seems to be working a treat.  Cheers @l2isbad, appeciated

**Comment by l2isbad** [comment]
@ktsaou ready

**Comment by ktsaou** [comment]
merged it. Thank you!

**Pull request merged**